### PR TITLE
Presentation: Unique values in instance filter should only be retrieved from instances that are child nodes for filtered parent node

### PR DESCRIPTION
**Pull request by medikornov**
Closes: https://github.com/iTwin/itwinjs-core/issues/5725
imodel-native: https://github.com/iTwin/imodel-native/tree/presentation/unique-values-based-on-parent-node
imodel-native PR: https://github.com/iTwin/imodel-native/pull/364

### CODE REVIEW

File: presentation/common/src/presentation-common/content/Descriptor.ts [hunk 1]
```diff
@@ -326,6 +335,12 @@ export class Descriptor implements DescriptorSource {
   public readonly fields: Field[];
   /** [[ContentFlags]] used to create the descriptor */
   public readonly contentFlags: number;
+  /**
+   * A ruleset used to create the nodes descriptor for a specific hierarchy level.
+   * It is only returned with `GetNodesDescriptor`. Otherwise, undefined.
+   * @beta
+   */
+  public readonly hierarchyLevelRuleset?: Ruleset;
```
[the remark this answers is not in the document]
Comment (grigasp) [reply presentation/common/src/presentation-common/content/Descriptor.ts:343]: Also, I believe the new attribute should be (de)serialized in the `toJSON` and `fromJSON`.

File: full-stack-tests/presentation/src/Utils.ts [hunk 2]
```diff
@@ -111,6 +111,57 @@ export function insertDocumentPartition(db: IModelDb, code: string, label?: stri
   return { className: "BisCore:DocumentPartition", id };
 }
 
+/** Insert a model element into created imodel. Return created element's className and Id. */
+export function insertPhysicalModel(db: IModelDb, label: string, parentId?: Id64String) {
```
Comment (grigasp) [inline full-stack-tests/presentation/src/Utils.ts:115]: Let's move these iModel setup functions into a separate source file, e.g. `IModelSetupUtils.ts` or something like that.

File: full-stack-tests/presentation/src/frontend/Content.test.ts [hunk 3]
```diff
@@ -273,12 +274,97 @@ describe("Content", () => {
...
+              groupByLabel: false,
+            },
+          ],
+        }],
+      };
+      const rootNodes = await Presentation.presentation.getNodes({ imodel: testIModel, rulesetOrId: ruleset });
+      expect(rootNodes.length).to.eq(2);
+
+      const descriptor = await Presentation.presentation.getNodesDescriptor({ imodel: testIModel, rulesetOrId: ruleset, parentKey: rootNodes[0].key });
+      // ensure descriptor returns a correct ruleset
```
Comment (grigasp) [inline full-stack-tests/presentation/src/frontend/Content.test.ts:332]:
```suggestion

      // ensure descriptor contains the ruleset used to create it
```

File: full-stack-tests/presentation/src/frontend/Content.test.ts [hunk 4]
```diff
@@ -273,12 +274,97 @@ describe("Content", () => {
...
+      expect(descriptor!.hierarchyLevelRuleset!.rules).to.deep.equal([{
```
Comment (grigasp) [inline full-stack-tests/presentation/src/frontend/Content.test.ts:333]: you could avoid `descriptor!` by adding `assert(descriptor)` after retrieving it

**Review by grigasp — CHANGES_REQUESTED** [review]

File: presentation/common/src/presentation-common/content/Descriptor.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (medikornov) [reply presentation/common/src/presentation-common/content/Descriptor.ts]: Added serialization in `toJson`.  `fromJson` serializes ruleset implicitly.

File: full-stack-tests/presentation/src/Utils.ts (hunk 2 above)
Comment (medikornov) [reply full-stack-tests/presentation/src/Utils.ts]: moved

File: full-stack-tests/presentation/src/frontend/Content.test.ts (hunk 4 above)
Comment (medikornov) [reply full-stack-tests/presentation/src/frontend/Content.test.ts]: added

**Review by grigasp — APPROVED** [review]

**Pull request merged**